uaclient/entitlements/fips.py

```diff
@@ -167,6 +170,46 @@ def install_packages(
         :param cleanup_on_failure: Cleanup apt files if apt install fails.
         :param verbose: If true, print messages to stdout
         """
+
+        # Prior to installing packages, check if the kernel is being downgraded
+        # and if so verify that the user wants to continue
+        our_full_kernel_str = (
+            system.get_kernel_info().proc_version_signature_version
+        )
+        if our_full_kernel_str is None:
+            LOG.warning("Cannot gather kernel information")
+            return
+        our_m = re.search(
+            r"(?P<kernel_version>\d+\.\d+\.\d+)", our_full_kernel_str
+        )
+        fips_kernel_policy = apt.get_apt_cache_policy_for_package("linux-fips")
```

Comment:
Maybe I am missing something here, but can't we use `apt.get_pkg_version` here ?

Reply:
Does `apt.get_pkg_version` return the currently installed version on the to-be-installed version?  I would be much happier to use something cleaner if it exists.

Reply:
`apt.get_pkg_version` -> installed version
`apt.get_pkg_candidate_version` -> to-be-installed (candidate) version